### ISSUE TITLE
For Noah-MP, the vegetation fraction in output files is not the same values as used by the model.

### ISSUE THREAD

**Issue by ClaraDraper-NOAA** [NOAA-EMC/ufsatm#512]
## Description

Currently, when using the Noah-MP model, the vegetation fraction is set during each Noah-MP call, with the exact way it is set depending on the iopt_dveg setting. The vegetation fraction (fveg) is used in Noah-MP, and then passed out to the noah-mp driver as vegetation_fraction, where it is used. However, this value is not saved to the sfcprop%vfrac variable, which has a different set of data. sfcprop%vfrac then gets written out as veg in the output files.  

This creates a risk that the veg / vfrac data will be misunderstood (it also requires the data user to know where to look in the output file to find the actual vegetation fraction that was used, depending on the Noah-MP settings). This could be solved by: 

-writing the vegetation_fraction in the Noah-MP driver to the sfcprop%vfrac
-at a minumum, putting no data values in sfcprop%vfrac so that people know not to use it.

### To Reproduce:

Using control_p8 test, compare the output veg (not used) to the output shdmax (used as vegetation fraction with current Noah-MP defaults).

**Comment by ClaraDraper-NOAA** [comment]
@barlage  - I don't seem to have permission to assign people.

**Comment by HelinWei-NOAA** [comment]
sfcprop%vfrac refers to seasonal varying GVF (Green Vegetation Fraction). Both Noah and LUC LSM use it to reflect the vegetation seasonality while LAI is set o the constant value. While Noah-MP uses the varied LAI for vegetation seasonality and annal maximum GVF is used to indicate the vegetation cover fraction. So sfcprop%vfrac has the different definition for Noah-MP compared to Noah and LUC.

**Comment by ClaraDraper-NOAA** [comment]
Thanks @HelinWei-NOAA Am I correct though that sfcprop%vfrac is not actually used in Noah-MP? If it is used, can you please tell me where (if it is used, I need to change the land perturbation scheme for Noah-MP).

**Comment by HelinWei-NOAA** [comment]
We recommend using option 4 for dveg. In this case sfcprop%vfrac  (shdfac in the code) is not used in Noah-MP. Instead we use shdmax (maximum shdfac) as fveg.

**Comment by HelinWei-NOAA** [comment]
For Noah-MP, it makes more sense to perturb LAI+SAI rather than vfrac or fveg.

**Comment by ClaraDraper-NOAA** [comment]
@HelinWei-NOAA You say:  " We recommend using option 4 for dveg. In this case sfcprop%vfrac (shdfac in the code) is not used in Noah-MP."  This is the point I'm trying to make. If we're not vfrac, I don't think it should be included in the model output, as this will confuse people (like it did me!).

**Comment by junwang-noaa** [comment]
@ClaraDraper-NOAA @HelinWei-NOAA It looks to me this is a diag_table issue that the veg (sfcprop%vfrac) should not be output if Noah-MP is used, is it correct? Do we have Noah-MP documentation that we can add this information? Also do we still want to update the code to have undefined values for veg when using NoahMP?

**Comment by ClaraDraper-NOAA** [comment]
@junwang-noaa Yes, I suggest that when Noah-MP is used with dveg options that do not use the sfcprop%vfrac we should not output the sfcprop%vfrac (or output NaN-type values) as it's misleading. I don't know if there are any dveg options that actually use sfcprop%vfrac though.
